pkg/apis/core/validation/validation.go

```diff
@@ -5597,6 +5597,11 @@ func ValidateNamespace(namespace *core.Namespace) field.ErrorList {
 	for i := range namespace.Spec.Finalizers {
 		allErrs = append(allErrs, validateFinalizerName(string(namespace.Spec.Finalizers[i]), field.NewPath("spec", "finalizers"))...)
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.NamespaceDefaultLabelName) {
+		if label, ok := namespace.Labels[v1.LabelMetadataName]; len(namespace.Name) > 0 && ok && label != namespace.Name {
```

Comment:
checking `ok` means a missing label is permitted. simplify to this and add a test that would have caught this:

```suggestion
		if len(namespace.Name) > 0 && namespace.Labels[v1.LabelMetadataName] != namespace.Name {
```

Reply:
The problem here is actually on the namespace strategy_test, that calls the PrepareForCreate and then the [Validate](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/namespace/strategy_test.go#L50)

Because Validate will expect that the Defaulting will add the label, but apparently the Strategy does not follow the Defaulting, it will fail. So just to close this question, what would be better to do here?
* On strategy_test.go in that part of the code call Canonicalize() before Validation?
* Before PrepareForCreate, on the Object specification add also the label?

I'm moving with adding the Label to the Object specification, as this seems the better approach to me, but if you think another approach should be taken please let me know! :)

Once you're ok with the approach, I can squash the commits (or maybe add the tide/merge-method-squash if this is something desired!)

Thanks for your time Jordan :)